testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDatabricksUnityCompatibility.java

```diff
@@ -0,0 +1,213 @@
+                .containsOnly(expectedRowsForInsert);
+
+        // update
+        List<QueryAssert.Row> expectedRowsForUpdate = ImmutableList.of(row(1, "one"), row(2, "two hundred"));
+        onTrino().executeQuery(format("UPDATE delta.%s.%s SET c2 = 'two hundred' WHERE c1 = 2", schemaName, tableName));
+        assertThat(onTrino().executeQuery(format("SELECT * FROM delta.%s.%s", schemaName, tableName)))
+                .containsOnly(expectedRowsForUpdate);
+        assertThat(onDelta().executeQuery(format("SELECT * FROM %s.%s.%s", unityCatalogName, schemaName, tableName)))
+                .containsOnly(expectedRowsForUpdate);
+
+        // delete
+        List<QueryAssert.Row> expectedRowsForDelete = ImmutableList.of(row(2, "two hundred"));
+        onTrino().executeQuery(format("DELETE FROM delta.%s.%s WHERE c2 = 'one'", schemaName, tableName));
+        assertThat(onTrino().executeQuery(format("SELECT * FROM delta.%s.%s", schemaName, tableName)))
+                .containsOnly(expectedRowsForDelete);
+        assertThat(onDelta().executeQuery(format("SELECT * FROM %s.%s.%s", unityCatalogName, schemaName, tableName)))
+                .containsOnly(expectedRowsForDelete);
+
+        // merge
+        List<QueryAssert.Row> expectedRowsForMerge = ImmutableList.of(row(1, "one"), row(2, "two"), row(3, "three"));
+        String sourceTableName = "test_source_" + randomNameSuffix();
+        String tableLocation2 = format("%s/%s/%s", externalLocationPath, schemaName, sourceTableName);
+        onDelta().executeQuery(
+                format("CREATE TABLE %s.%s.%s (c1 int, c2 string) using delta location '%s'",
+                        unityCatalogName, schemaName, sourceTableName, tableLocation2));
+        onDelta().executeQuery(
+                format("INSERT INTO %s.%s.%s values (1, 'one'), (2, 'two'), (3, 'three')",
+                        unityCatalogName, schemaName, sourceTableName));
+
+        onTrino().executeQuery(format("MERGE INTO delta.%s.%s t USING delta.%s.%s s on t.c1 = s.c1 " +
+                "WHEN MATCHED THEN UPDATE SET c2 = s.c2 " +
+                "WHEN NOT MATCHED THEN INSERT (c1, c2) VALUES (s.c1, s.c2)", schemaName, tableName, schemaName, sourceTableName));
+        assertThat(onTrino().executeQuery(format("SELECT * FROM delta.%s.%s", schemaName, tableName)))
+                .containsOnly(expectedRowsForMerge);
+        assertThat(onDelta().executeQuery(format("SELECT * FROM %s.%s.%s", unityCatalogName, schemaName, tableName)))
+                .containsOnly(expectedRowsForMerge);
+    }
+
+    @Test(groups = {DATABRICKS_UNITY_HTTP_HMS, PROFILE_SPECIFIC_TESTS})
+    @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
+    public void testManagedSchemas()
+    {
+        assertThat(onTrino().executeQuery(format("SHOW CREATE SCHEMA delta.%s", schemaName)))
+                .containsOnly(row(format("CREATE SCHEMA delta.%s", schemaName)));
+    }
+
```

Comment:
gentle reminder

Reply:
Sorry in case I am misunderstanding what you mean. The managed tables are not exposed via the API at this moment. So there is no way read them from Trino currently. I updated the test to assert that a managed table created from databricks is not visible in the `show tables` on trino. Is that not what you meant?

Reply:
"select * from delta.managedschema.managedtable" will fail with a different exception than table not found exception.

Reply:
Discussed with @findinpath on Slack. this is not necessary because the test assumes that user know the table name before hand. We can potentially fix it on the Unity Catalog service side to throw a TableNotFoundException in such cases, but that is out of scope of this PR.